### PR TITLE
Revive DefaultRoslynCompilationServiceTest

### LINKED ISSUE #4140: Revive DefaultRoslynCompilationServicesTest
We had tests for the DNX based RoslynCompilationService, but not the DependencyModel based compilation service. We need to revive these.

### PR DESCRIPTION
**Pull request by pranavkm**
Partially fixes #4140

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs [hunk 1]
```diff
@@ -238,16 +249,30 @@ private static string GetFilePath(string relativePath, Diagnostic diagnostic)
...
+
+            for (var i = 0; i < _dependencyContext.CompileLibraries.Count; i++)
+            {
+                var library = _dependencyContext.CompileLibraries[i];
+                IEnumerable<string> referencePaths;
+                try
+                {
+                    referencePaths = library.ResolveReferencePaths();
+                }
+                catch (InvalidOperationException)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: Workaround for https://github.com/dotnet/cli/issues/1694.

**Comment by pranavkm** [comment]
cc @NTaylorMullen

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+            Assert.Equal(relativeFileInfo.RelativePath, compilationFailure.SourceFilePath);
+            Assert.Equal(fileContent, compilationFailure.SourceFileContent);
+        }
+
+        [Fact]
+        public void Compile_ReturnsGeneratedCodePath_IfLinePragmaIsNotAvailable()
+        {
+            // Arrange
+            var fileContent = "file content";
+            var content = @"this should fail";
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs]: no need for `@`

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+            // Act
+            var result = compilationService.Compile(relativeFileInfo, content);
+
+            // Assert
+            Assert.IsType<CompilationResult>(result);
+            Assert.Null(result.CompiledType);
+
+            var compilationFailure = Assert.Single(result.CompilationFailures);
+            Assert.Equal("Generated Code", compilationFailure.SourceFilePath);
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs]: Confused, why does this not return the relative file info's path?

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+            return new DiagnosticDescriptor(
+                id: "someid",
+                title: "sometitle",
+                messageFormat: messageFormat,
+                category: "some-category",
+                defaultSeverity: DiagnosticSeverity.Error,
+                isEnabledByDefault: true);
+        }
+
+        private static RazorViewEngineOptions GetOptions(Action<RoslynCompilationContext> callback = null)
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs]: Not needed, just inline `new RazorViewEngineOptions()` wherever you have `GetOptions`. The options class handles the `null` callback case.

**Comment by NTaylorMullen** [comment]
:watch: just for questions

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs (hunk 3 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs]: This is some quirky behavior related to - https://github.com/dotnet/roslyn/issues/1920. If we return the `SourceFilePath` here, it ends up not allowing breakpoints to be set unless you have the Allow source files to be different... flag set in the VS.

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs (hunk 2 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs]: Sure.

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs (hunk 4 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/DefaultRoslynCompilationServiceTest.cs]: :+1:

**Comment by NTaylorMullen** [comment]
:shipit: questions answered

**Pull request closed without merging**